src/utils.cpp

```diff
@@ -355,9 +355,51 @@ std::string utils::retrieve_url(const std::string& url,
 		curl_easy_setopt(easyhandle, CURLOPT_USERPWD, authinfo.c_str());
 	}
 
-	curl_easy_perform(easyhandle);
+	// Error handling as per https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html
+	char errbuf[CURL_ERROR_SIZE];
+	// Please note that we clobber CURLOPT_ERRORBUFFER here in case of cached handles
+	// Currently, this is the only place we do this, so this should be fine.
+	// There does not seem to be a way to query curlopts, so we cannot save the old errorbuf value here...
+	curl_easy_setopt(easyhandle, CURLOPT_ERRORBUFFER, errbuf);
+	errbuf[0] = '\0';
+
+	CURLcode res = curl_easy_perform(easyhandle);
+
+	// Report curl errors
+	if (res != CURLE_OK) {
+		std::string errmsg(errbuf);
+		if (errmsg.empty()) {
+			errmsg = curl_easy_strerror(res);
```

Comment:
Does `curl_easy_strerror` append `\n`? The man page doesn't say.

If it doesn't, "prettification" should be done always.

Reply:
Ah, good catch! Actually, from the example [here](https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html), I would assume that it actually never does :)

Reply:
hmm, but apparently our Logging routine already appends "\n" unconditionally, so the correct prettyfication is actually the reverse (dropping superfluous '\n's)